airflow/api_connexion/schemas/dag_run_schema.py

```diff
@@ -104,6 +105,12 @@ def autofill(self, data, **kwargs):
         return data
 
 
+class SetDagRunStateFormSchema(Schema):
```

Comment:
Can we use DAGRunSchema schema here? I think it is possible, but we need to carefully review which fields are marked as input and which are marked as output.

Reply:
Using `PATCH` is a good idea though. I was leaning toward `POST` with the impression `dagrun.state = new_state` does not work, but I was wrong, it actually implicitly calls `set_state`.

Reply:
considering @uranusjr 's comment, I'll resolve this conversation without changing the super class here (let me know if you still want it to be done)